src/BuiltInTools/dotnet-watch/HotReload/CompilationWorkspaceProvider.cs

```diff
@@ -65,5 +77,23 @@ static async void CreateProject(TaskCompletionSource<(Solution, WatchHotReloadSe
 
             taskCompletionSource.TrySetResult((currentSolution, hotReloadService));
         }
+
+        private static async Task<ImmutableArray<string>> GetHotReloadCapabilitiesAsync(Task<ImmutableArray<string>> hotReloadCapabilitiesTask, IReporter reporter)
+        {
+            try
+            {
+                var capabilities = await hotReloadCapabilitiesTask;
+                reporter.Verbose($"Hot reload capabilities: {string.Join(" ", capabilities)}.");
+
+                return capabilities;
+            }
+            catch (Exception ex)
+            {
+                reporter.Verbose("Reading hot reload capabilities failed. Using default capabilities.");
+                reporter.Verbose(ex.ToString());
+
+                return ImmutableArray.Create("Baseline", "AddDefinitionToExistingType", "NewTypeDefinition");
```

Comment:
What is this doing?

Reply:
I _think_ the compiler expects some baseline capabilities. These were what we previously had here, so I'm just returning them in case we can't get it from the app.